android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/views/InstructionsView.kt

```diff
@@ -36,23 +49,60 @@ fun InstructionsView(
     distanceToNextManeuver: Double?,
     distanceFormatter: DistanceFormatter = LocalizedDistanceFormatter(),
     theme: InstructionRowTheme = DefaultInstructionRowTheme,
-    content: @Composable () -> Unit = {
-      ManeuverImage(instructions.primaryContent, tint = MaterialTheme.colorScheme.primary)
+    remainingSteps: List<RouteStep>? = null,
+    initExpanded: Boolean = false,
+    contentBuilder: @Composable (VisualInstruction) -> Unit = {
+      ManeuverImage(it.primaryContent, tint = MaterialTheme.colorScheme.primary)
     }
 ) {
-  Column(
+  var isExpanded by remember { mutableStateOf(initExpanded) }
+  val screenHeight = LocalConfiguration.current.screenHeightDp
+  val collapsedHeight = 100.dp
+
+  val targetHeight = if (isExpanded) screenHeight.dp else collapsedHeight
+  val animatedHeight by animateDpAsState(targetValue = targetHeight)
+
+  val scrollState = rememberScrollState()
+
+  Box(
       modifier =
           Modifier.fillMaxWidth()
-              .shadow(elevation = 5.dp, RoundedCornerShape(10.dp))
+              .height(animatedHeight)
               .background(theme.backgroundColor, RoundedCornerShape(10.dp))
-              .padding(8.dp)) {
-        ManeuverInstructionView(
-            text = instructions.primaryContent.text,
-            distanceFormatter = distanceFormatter,
-            distanceToNextManeuver = distanceToNextManeuver,
-            theme = theme,
-            content = content)
-        // TODO: Secondary instructions
+              .clickable { isExpanded = !isExpanded }
+              .padding(16.dp)) {
+        Column(modifier = if (isExpanded) Modifier.verticalScroll(scrollState) else Modifier) {
+          // Primary content
+          ManeuverInstructionView(
+              text = instructions.primaryContent.text,
+              distanceFormatter = distanceFormatter,
+              distanceToNextManeuver = distanceToNextManeuver,
+              theme = theme) {
+                contentBuilder(instructions)
+              }
+
+          // TODO: Secondary content
+
+          // Expanded content
+          if (isExpanded && remainingSteps != null && remainingSteps.count() > 1) {
+            Spacer(modifier = Modifier.height(8.dp))
+            HorizontalDivider(thickness = 1.dp)
+            remainingSteps.drop(1).forEach { step ->
+              step.visualInstructions.firstOrNull()?.let { upcomingInstruction ->
+                Spacer(modifier = Modifier.height(8.dp))
+                ManeuverInstructionView(
+                    text = upcomingInstruction.primaryContent.text,
+                    distanceFormatter = distanceFormatter,
+                    distanceToNextManeuver = step.distance,
+                    theme = theme) {
+                      contentBuilder(upcomingInstruction)
+                    }
+                Spacer(modifier = Modifier.height(8.dp))
+                HorizontalDivider(thickness = 1.dp)
+              }
+            }
```

Comment:
Around here we probably also want the footer w/ a close like on iOS? Whatever "arrow up" option is included in the material icons we've included. This would make the lazy column look like:

```kotlin
LazyColumn(
    verticalArrangement = Arrangement.spacedBy(16.dp)
) {
    items(remainingSteps.drop(1)) { step ->
        ManeuverInstructionView(step)
        HorizontalDivider(thickness = 1.dp) // interestingly most android tables seem to omit dividers these days.
    }

    item { 
         // on click could be omitted given the whole list clicks, but may be handy if someone doesn't want tap anywhere.
         InstructionsFooter(onClick: { isExpanded = !isExpanded }) 
    }
}
```

Reply:
![image](https://github.com/user-attachments/assets/3b22d5fe-fb98-4ef4-814d-c4a3719f57aa)

How do you feel about this visual? I'm not sold on the button styling myself since it's pretty heavy, but it's how material theming works out of the box, and it feels like getting into a war if we want to fight it :/

Reply:
Updated video:

https://github.com/user-attachments/assets/2d781587-4559-4049-befb-0dc7e5e3d06d

Reply:
Discussed over Slack; found something better.

![image](https://github.com/user-attachments/assets/31f0d3cb-de8e-449c-b6b8-d43d3ee5f703)